### ISSUE TITLE
StaticFile.fromFile IllegalArgumentException with large files

### ISSUE THREAD

**Issue by lukestephenson** [http4s/http4s#1554]
The following code results in an IllegalArgumentException when the file is longer than `Int.MaxValue`
```
case req@GET -> Root / "file" =>
      StaticFile.fromFile(new File("massive-file.txt"), Some(req)).getOrElseF(NotFound())
```

And the exceptionis
```
logtype="application" transaction_id="" req_time="2017-11-15T10:53:34.195+11:00" loglevel="ERROR" threadname="scala-execution-context-global-73" classname="o.h.b.p.Stage" linenumber="Http1ServerStage.scala:226" message="Error writing body"
java.lang.IllegalArgumentException: null
	at java.nio.ByteBuffer.allocate(ByteBuffer.java:334)
	at fs2.io.file.FileHandle$$anon$2.read(FileHandle.scala:160)
	at org.http4s.StaticFile$._readAllFromFileHandle0$1(StaticFile.scala:150)
	at org.http4s.StaticFile$.readAllFromFileHandle$1(StaticFile.scala:145)
	at org.http4s.StaticFile$.$anonfun$fileToBody$7(StaticFile.scala:160)
	at org.http4s.StaticFile$.$anonfun$fileToBody$7$adapted(StaticFile.scala:160)
```

----

Some background as to how / why I discovered this. I was looking into streaming of large responses and noticed that the JVM CPU usage is very high while doing so.  I wasn't expecting this as I expect that Streaming of a large response is largely IO.  To narrow down to the least possible application code I decided just to stream a large file.  CPU usage is still higher than what I'd expect.  I'm still investigating.

**Comment by ChristopherDavenport** [comment]
Is this the code that has to unzip the file from the jar?

**Comment by lukestephenson** [comment]
Nope.  The file is just being read directly.

**Comment by rossabaker** [comment]
That exception happens when you allocate a negative buffer size.  I bet the [`.toInt` is the villain of this story](https://github.com/http4s/http4s/blob/b435b6be66392b80e12dd9bdbe2d0b20b83f7807/core/src/main/scala/org/http4s/StaticFile.scala#L150).

**Comment by rossabaker** [comment]
I think that `(end - offset)` is to make sure we don't try to read beyond the end of the file.  I think if we min that against 0, we'll be fine.

I'm also don't see anything that prevents trying to read past the end of the file if a `start` and `end` are given that are out of bounds.

Does anyone want to try a PR?  It looks like this could be fixed starting in release-0.17.x.

**Comment by rossabaker** [comment]
I fixed this while looking at #1555. PR coming.
